torch/autograd/gradcheck.py

```diff
@@ -223,7 +227,7 @@ def _combine_jacobian_cols(jacobians_cols: Dict[int, List[torch.Tensor]], output
                            numel) -> Tuple[torch.Tensor, ...]:
     # jacobian_cols maps column_idx -> output_idx -> single column of jacobian Tensor
     # we return a list that maps output_idx -> full jacobian Tensor
-    jacobians = _allocate_jacobians_with_outputs(outputs, numel, input.dtype, input.device)
+    jacobians = _allocate_jacobians_with_outputs(outputs, numel, dtype=input.dtype if input.dtype.is_complex else None)
```

Comment:
Previously this function allocated tensors to have the dtype and device of the inputs (but after the change, the outputs), so not sure if you meant to make this change.

Reply:
This function was always called with real valued outputs. And so the dtype of the input was always the thing to use.
Now when the input is real but output is complex, we want to use the output dtype.
So this doesn't change the behavior of existing code, but allows for complex output.